process/transaction/shardProcess.go

```diff
@@ -669,6 +685,37 @@ func (txProc *txProcessor) addFeeAndValueToDest(acntDst state.UserAccountHandler
 	return txProc.accounts.SaveAccount(acntDst)
 }
 
+func (txProc *txProcessor) processRelayedTxV3(tx *transaction.Transaction) (vmcommon.ReturnCode, error) {
+	relayerAccount, sndAccount, err := txProc.getAccounts(tx.RelayerAddr, tx.SndAddr)
+	if err != nil {
+		return 0, err
+	}
+
+	if !txProc.enableEpochsHandler.IsFlagEnabled(common.RelayedTransactionsV3Flag) {
+		return vmcommon.UserError, txProc.executingFailedTransaction(tx, relayerAccount, process.ErrRelayedTxV3Disabled)
+	}
+
+	if !txProc.shardCoordinator.SameShard(tx.RelayerAddr, tx.SndAddr) {
+		return vmcommon.UserError, txProc.executingFailedTransaction(tx, relayerAccount, process.ErrShardIdMissmatch)
+	}
+
+	if !check.IfNil(relayerAccount) && relayerAccount.IsGuarded() {
+		return vmcommon.UserError, txProc.executingFailedTransaction(tx, relayerAccount, process.ErrGuardedRelayerNotAllowed)
+	}
+
+	userTx := *tx
+	// remove relayer addr and signature for tx type handler
+	userTx.RelayerAddr = nil
```

Comment:
it could work without these 2 lines.

Reply:
with the current implementation of [ProcessTransactionType](https://github.com/multiversx/mx-chain-go/blob/relayedv3/process/coordinator/transactionType.go#L80) from txTypeHandler, if these fields won't be removed, the user tx will still be considered of type relayed v3

at [this point](https://github.com/multiversx/mx-chain-go/blob/relayedv3/process/transaction/shardProcess.go#L913), we want to get the proper type of the user tx for its processing

Reply:
yeah. but this is not ok at all. somebody make a hash for the USER TX - and it gets completely different, links will be lost.

You may create a new function down the line - which works a little bit differently. 

I would also separate into new function because we want to get rid of relayedV1 and relayedV2.

Reply:
pushed the suggestions + improved the comment of mentioned area